Source/Scene/Cesium3DTilesetTraversal.js

```diff
@@ -0,0 +1,823 @@
+                    } else {
+                        allReady = allReady && this.internalDFS.execute(child);
+                    }
+                }
+            }
+
+            if (allReady) {
+                return children;
+            }
+        }
+        return emptyArray;
+    };
+
+    function baseUpdateAndCheckChildren(tileset, tile, baseScreenSpaceError, frameState) {
+        if (tile.hasTilesetContent) {
+            // load any tilesets of tilesets now because at this point we still have not achieved a base level of content
+            loadTile(tile, frameState);
+            if (tile.contentReady) {
+                updateChildren(tileset, tile, frameState);
+            }
+            return true;
+        }
+
+        if (hasAdditiveContent(tile)) {
+            tileset._desiredTiles.push(tile);
+        }
+
+        // stop traversal when we've attained the desired level of error
+        if (tile._screenSpaceError <= baseScreenSpaceError) {
+            // When skipping LODs, require an existing base level of content first
+            updateChildren(tileset, tile, frameState);
+            return false;
+        }
+
+        var childrenVisibility = updateChildren(tileset, tile, frameState);
+        var showAdditive = tile.refine === Cesium3DTileRefine.ADD;
+        var showReplacement = tile.refine === Cesium3DTileRefine.REPLACE && (childrenVisibility & Cesium3DTileChildrenVisibility.VISIBLE_IN_REQUEST_VOLUME) !== 0;
+
+        return showAdditive || showReplacement || tile.hasTilesetContent || !defined(tile._ancestorWithContent);
+    }
+
+    BaseTraversal.prototype.shouldVisit = function(tile) {
+        return isVisible(tile.visibilityPlaneMask);
+    };
+
+    BaseTraversal.prototype.leafHandler = function(tile) {
+        // if skipLODs is off, leaves of the base traversal get pushed to tileset._desiredTiles. additive tiles have already been pushed
+        if (this.tileset.skipLODs || !hasAdditiveContent(tile)) {
+            if (tile.refine === Cesium3DTileRefine.REPLACE && !childrenAreVisible(tile)) {
+                ++this.tileset._statistics.numberOfTilesCulledWithChildrenUnion;
+                return;
+            }
+            this.leaves.push(tile);
+        }
+    };
+
+    function InternalBaseTraversal(options) {
+        this.tileset = undefined;
+        this.frameState = undefined;
+        this.outOfCore = undefined;
+        this.baseScreenSpaceError = undefined;
+        this.stack = new ManagedArray();
+    }
+
+    InternalBaseTraversal.prototype.execute = function(root) {
+        this.allLoaded = true;
+        DFS(root, this);
+        return this.allLoaded;
+    };
+
+    InternalBaseTraversal.prototype.visitStart = function(tile) {
+        if (tile._lastVisitedFrame !== this.frameState.frameNumber) {
+            visitTile(this.tileset, tile, this.frameState, this.outOfCore);
+        }
+        if (!tile.contentReady) {
+            this.allLoaded = false;
+        }
```

Comment:
Since `shouldVisit()` will return false for any tile with renderable content, this code would not got reached for those tiles. This would create a situation `allLoaded` is true even if one of its descendants with renderable content is not ready.